lib/pavilion/test_run.py

```diff
@@ -436,6 +437,16 @@ def build(self):
 :return: True if these steps completed successfully.
 """
 
+        if self.config['build']['source_download_name'] is not None:
```

Comment:
Move this to __init__, and raise an error. That way we can immediately see that there's a problem.